src/App.tsx

```diff
@@ -1,15 +1,31 @@
-import React from 'react';
+import React, { useContext } from "react";
 import { Routes, Route, Link } from "react-router-dom";
-import './App.css';
-import HomeView from './views/HomeView';
-import RecipeView from './views/RecipeView';
-import RecipesView from './views/RecipesView';
+import "./App.css";
+import HomeView from "./views/HomeView";
+import RecipeView from "./views/RecipeView";
+import RecipesView from "./views/RecipesView";
+import { useQuery } from "@tanstack/react-query";
+import getUser from "./api/getUser";
+import { AppContext } from "./store/app";
+import "./i18n";
 
 function App() {
+  const { setUser } = useContext(AppContext);
+
+  const { isLoading } = useQuery({
+    queryKey: ["user"],
+    queryFn: getUser,
+    onSuccess: setUser,
+  });
+
+  if (isLoading) return <div>User Loading...</div>;
```

Comment:
User loading indicator can be improved some how, but this nicely shows us the progress of UI states.